x-pack/plugins/ingest_manager/server/services/agent_config.ts

```diff
@@ -83,10 +83,7 @@ class AgentConfigService {
       return this.create(soClient, newDefaultAgentConfig);
     }
 
-    return {
-      id: configs.saved_objects[0].id,
-      ...configs.saved_objects[0].attributes,
-    };
+    return configs.saved_objects[0].attributes;
```

Comment:
here and below `id` is always present in `attributes` types ==> `attribute.id` always rewrite `id`. feel free to revert the changes.